### PR TITLE
Done: Login screen

### PR DESCRIPTION
**Pull request by gonzaloHT**
https://trello.com/c/FzDoFSwo/1-create-login-screen

### CODE REVIEW

File: theworkoutinfo/Managers/AuthenticationManager.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    var trySigning: AnyObserver<AuthenticationCredentials>
+    var signInResult: Observable<LoginResult>
+    
+    init() {
+        let _trySigning = PublishSubject<AuthenticationCredentials>()
+        trySigning = _trySigning.asObserver()
+        signInResult = _trySigning.asObservable().map { (authCredentials) -> LoginResult in
+            let userCredentials = AuthenticationHelper.getUserCredentials()
+            let credentialsOk = (authCredentials.email == userCredentials.email && authCredentials.password == userCredentials.password)
+            switch credentialsOk {
```
Comment (gastonkosut) [inline theworkoutinfo/Managers/AuthenticationManager.swift]: Switch for true/false is not needed, let's use just a if

File: theworkoutinfo/Managers/AuthenticationManager.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    static let sharedInstance = AuthenticationManager()
+    
+    var trySigning: AnyObserver<AuthenticationCredentials>
+    var signInResult: Observable<LoginResult>
+    
+    init() {
+        let _trySigning = PublishSubject<AuthenticationCredentials>()
+        trySigning = _trySigning.asObserver()
+        signInResult = _trySigning.asObservable().map { (authCredentials) -> LoginResult in
+            let userCredentials = AuthenticationHelper.getUserCredentials()
```
Comment (gastonkosut) [inline theworkoutinfo/Managers/AuthenticationManager.swift:37]: Let's create a Comparable to override the '==' for the AuthenticationCredentials objects

File: theworkoutinfo/Managers/AuthenticationManager.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            let credentialsOk = (authCredentials.email == userCredentials.email && authCredentials.password == userCredentials.password)
+            switch credentialsOk {
+            case true:
+                return LoginResult.success
+            case false:
+                let result: LoginResult
+                (authCredentials.email == userCredentials.email) ? (result = LoginResult.wrongPassword) : (result = LoginResult.wrongEmail)
```
Comment (gastonkosut) [inline theworkoutinfo/Managers/AuthenticationManager.swift]: It's not secure to give the user the information about what field failed. We can discuss this

File: theworkoutinfo/ViewControllers/LoginViewController.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        viewModel.isValid
+            .bind(to: loginButton.rx.isEnabled)
+            .disposed(by: disposeBag)
+        
+        loginButton.rx.tap
+            .bind(to: viewModel.tryLogin)
+            .disposed(by: disposeBag)
+        
+        viewModel.errorMessage.do(onNext: { errorMessage in
+            let alert = UIAlertController(title: "Error", message: errorMessage, preferredStyle: UIAlertControllerStyle.alert)
```
Comment (gastonkosut) [inline theworkoutinfo/ViewControllers/LoginViewController.swift]: Let's create an alert helper to use just something like AlertHelper.showMessage or showAlert(...

Comment (martinzp) [reply theworkoutinfo/ViewControllers/LoginViewController.swift]: It could be an extension for UIViewController too maybe

**Review by gastonkosut — APPROVED** [review]
👍

**Pull request merged**